### PR TITLE
Allowing SelectColumns to take a bare list of columns

### PR DESCRIPTION
**Pull request by rogancarr**
This PR overloads `SelectColumns` so that it can be used with a bare list of columns. The original API is changed to remove `keepHidden`, as this new API has the default behavior.

The PR also cleans up the sample, which had mismatches between the actual and expected outputs.

Fixes: #2371

### CODE REVIEW

**Comment by Ivanidzo4ka** [comment]
@artidoro Who I believe have same changes in his code.

File: src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs [hunk 1]
```diff
@@ -107,8 +107,28 @@ public static ColumnSelectingEstimator DropColumns(this TransformsCatalog catalo
...
+        /// <format type="text/markdown">
+        /// <see cref="SelectColumns(TransformsCatalog, string[])"/> operates on the schema of an input IDataView, dropping unselected columns from the schema.
+        /// </format>
+        /// </remarks>
+        /// <param name="catalog">The transform's catalog.</param>
+        /// <param name="keepColumns">The array of column names to keep.</param>
+        /// <example>
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        /// [!code-csharp[SelectColumns](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/SelectColumns.cs)]
```
Comment (TomFinley) [inline src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs]: >[!code-csharp[SelectColumns](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/SelectColumns.cs)] [](start = 12, length = 118)

Hi @rogancarr, just curious, do you have some sort of plugin or something that is helping you edit these without error? VS at least with its builtin facilities is giving me very little help here for my own work, just wondering how you are managing.

Comment (rogancarr) [reply src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs]: There are actually no debugging tools for these links at the moment. Just very careful coping and pasting and editing.

---
In reply to: [254457544](https://github.com/dotnet/machinelearning/pull/2439#discussion_r254457544) [](ancestors = 254457544)

File: src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs [hunk 2]
```diff
@@ -107,8 +107,28 @@ public static ColumnSelectingEstimator DropColumns(this TransformsCatalog catalo
         /// </example>
         public static ColumnSelectingEstimator SelectColumns(this TransformsCatalog catalog,
             string[] keepColumns,
-            bool keepHidden = ColumnSelectingTransformer.Defaults.KeepHidden)
+            bool keepHidden)
             => new ColumnSelectingEstimator(CatalogUtils.GetEnvironment(catalog),
                 keepColumns, null, keepHidden, ColumnSelectingTransformer.Defaults.IgnoreMissing);
+
+        /// <summary>
+        /// ColumnSelectingEstimator is used to select a list of columns that user wants to keep from a given input.
```
Comment (artidoro) [inline src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs]: Could you use the same comment across the two extensions?
Also I would remove the reference to user, maybe something like: 

```suggestion
        /// Select a list of columns to keep in a given <see cref="IDataView"/>.
```

Comment (rogancarr) [reply src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs]: Yes, good catch!

---
In reply to: [254465617](https://github.com/dotnet/machinelearning/pull/2439#discussion_r254465617) [](ancestors = 254465617)

**Pull request merged**